### ISSUE TITLE
[build] MacOS binary broken on Pyinstaller 5.9

### ISSUE THREAD

**Issue by pukkandan** [yt-dlp/yt-dlp#6541]
MacOS Univeral2 [builds fail](https://github.com/Grub4K/yt-dlp/actions/runs/4417464692/jobs/7743211308) with Pyinstaller 5.9.

For now, I have pinned Pyinstaller to 5.8, but we need a proper fix. After some experimentation, [not using the `/usr/bin/python3`](https://github.com/pukkandan/yt-dlp-dev/commit/78be0d84ee2c1863202bb16357ff3f7cd846e2e1) seems to [fix the problem](https://github.com/pukkandan/yt-dlp-dev/actions/runs/4419005400). However, I have no way to test whether [the artifact](https://nightly.link/pukkandan/yt-dlp-dev/actions/artifacts/598530203.zip) runs on various mac versions and architectures.

I will need someone to either test this build, or if it doesn't work, to investigate the issue further.

@smplayer-dev @StefanLobbenmeier

**Comment by throwawayay** [comment]
A data point: `yt-dlp_macos` from `archive.zip` confirmed working on Apple Silicon, macOS Ventura 13

**Comment by pukkandan** [comment]
Thanks. Apple silicon was the main thing that needed testing. Since the build is made on x86, it is unlikely that won't work

**Comment by StefanLobbenmeier** [comment]
Thanks 😄 i don’t have an Mac that old anymore unfortunately, will have to look into VMs eventually instead

**Comment by smplayer-dev** [comment]
The artifact seems to work fine on MacOS 10.15.

**Comment by dirkf** [comment]
So which Python is being run when this works correctly?

**Comment by Grub4K** [comment]
It should be this line from this [Release action](https://github.com/Grub4K/yt-dlp/actions/runs/4420187465/jobs/7749577384#step:5:8):
```
625 INFO: PyInstaller: 5.9.0
625 INFO: Python: 3.11.2
651 INFO: Platform: macOS-11.7.4-x86_64-i386-64bit
```

**Comment by bashonly** [comment]
Closed by 44a79958f0b596ee71e1eb25f158610aada29d1b
